okhttp/src/main/java/okhttp3/UnmodifiableList.java

```diff
@@ -0,0 +1,46 @@
+package okhttp3;
+
+import java.util.Iterator;
+import java.util.List;
+
+public final class UnmodifiableList<T> implements Iterable<T> {
```

Comment:
Opposed in general to this. This class doesn't carry its weight in an API
that we try very hard to minimize.

On Sat, Jan 9, 2016, 10:07 AM Artem Zinnatullin <notifications@github.com>
wrote:

> In okhttp/src/main/java/okhttp3/UnmodifiableList.java
> <https://github.com/square/okhttp/pull/2221#discussion_r49264805>:
>
> > @@ -0,0 +1,46 @@
> > +package okhttp3;
> > +
> > +import java.util.Iterator;
> > +import java.util.List;
> > +
> > +public final class UnmodifiableList<T> implements Iterable<T> {
>
> Not a fan of new public class for the project, but hadn't found a better
> solution yet. But, looks like it can be used not only for interceptors but
> also for other things too both internally and for the public API.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/square/okhttp/pull/2221/files#r49264805>.
>